src/test/java/study/StringTest.java

```diff
@@ -0,0 +1,94 @@
+package study;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.*;
+
+@DisplayName("String 테스트")
+public class StringTest {
+
+	@Test
+	@DisplayName("구분자가 있을 경우 split 테스트")
+	void splitContains() {
+		String str = "1,2";
+		String[] result = str.split(",");
+
+		assertThat(result).contains("1", "2");
+	}
+
+	@Test
+	@DisplayName("구분자가 없을 경우 split 테스트")
+	void splitContainsExactly() {
+		String str = "1";
+		String[] result = str.split(",");
+
+		assertThat(result).containsExactly("1");
+	}
+
+	@Test
+	@DisplayName("문자열에서 괄호문자 () 제거후 split 테스트")
+	void splitAfterRemoveBracket() {
+		String str = "(1,2)";
+		if (str.contains("(")) {
+			str = str.substring(str.indexOf("(") + 1);
+		}
+		if (str.contains(")")) {
+			str = str.substring(0, str.indexOf(")"));
+		}
+		String[] result = str.split(",");
+
+		assertThat(result).containsExactly("1", "2");
+	}
+
+	@Test
+	@DisplayName("charAt 테스트")
+	void charAt() {
+		String str = "abc";
+		char searchChar = str.charAt(0);
+
+		assertThat(searchChar).isEqualTo('a');
+	}
+
+	@Test
+	@DisplayName("charAt 예외발생 테스트1")
+	void charAtException1() {
+		String str = "abc";
+
+		assertThatThrownBy(() -> {
+
+			char searchChar = str.charAt(3);
+			assertThat(searchChar).isEqualTo('e');
+
+		}).isInstanceOf(IndexOutOfBoundsException.class)
+			.hasMessageContaining("String index out of range: 3");
+	}
```

Comment:
str.charAt(3) 에서 Exception 이 발생하기 때문에 이후 로직 실행되지 않으므로 제거해주는 것이 좋을 것 같아요! 😄 

또한 현재 charAt() 메소드에 대한 정상적인 테스트와 예외발생에 대한 테스트를 함께 시도해주신 것으로 보여지는데요
 [클린코드](https://velog.io/@youngerjesus/%ED%81%B4%EB%A6%B0-%EC%BD%94%EB%93%9C-Chapter-9.-%EB%8B%A8%EC%9C%84-%ED%85%8C%EC%8A%A4%ED%8A%B8)에서는 테스트 코드 하나당 한 개념만 테스트 하는 것을 권장합니다. 관련해서 한번 확인해보시면 좋을 것 같아요! 😄 
```suggestion
	@Test
    @DisplayName("charAt 예외발생 테스트1")
    void charAtException1() {
        String str = "abc";

        assertThatThrownBy(() -> str.charAt(3))
                .isInstanceOf(IndexOutOfBoundsException.class)
                .hasMessageContaining("String index out of range: 3");
    }
```

Reply:
실행되지 않을 코드를 남겨놓아서 혼란을 줄 여지가 있었군요. 좋으신 지적입니다. 그리고 하나의 테스트 메소드당 한개의 assert! 기억하겠습니다 😃

Reply:
링크주신 [클린코드](https://velog.io/@youngerjesus/%ED%81%B4%EB%A6%B0-%EC%BD%94%EB%93%9C-Chapter-9.-%EB%8B%A8%EC%9C%84-%ED%85%8C%EC%8A%A4%ED%8A%B8) 글도 잘 읽어보았습니다. 저에게 아주 유익한 정보였습니다. 👍